### PR TITLE
(#220) local threads lags fixes

### LINKED ISSUE #220: [REFACTORING] Take away ImageLoader.getFromDisk() method from the main thread
As I have already said in #215 the new method (ImageLoader.getFromDisk()) calls the File API on the main thread which may (and it does) cause lags when disk is busy by the other threads/processes. It shouldn't be hard to do it becase it's already kinda asynchronous (it returns ImageContainer which can be used for the request cancelling) and what is left to do is to jsut move the code onto some executor. 

Damn, I have just noticed that ImageLoader is a part of the Volley library. I probably shouldn't have changed it at all.

### PR DESCRIPTION
**Pull request by K1rakishou**
Closes #220 
We really need to get rid of volley and use some proper libraries for image loading like Glide.

### CODE REVIEW

**Comment by K1rakishou** [comment]
```
com.github.adamantcheese.chan, PID: 21177
    java.lang.NullPointerException: Attempt to invoke interface method 'void com.github.adamantcheese.chan.core.image.ImageListener.onErrorResponse(com.android.volley.VolleyError)' on a null object reference
        at com.github.adamantcheese.chan.core.image.ImageLoaderV2.lambda$null$0(ImageLoaderV2.java:117)
        at com.github.adamantcheese.chan.core.image.-$$Lambda$ImageLoaderV2$9FvfVKiM5vb6ewdwdUs4HVbO27g.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7343)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:933)
2019-07-30 12:52:23.855 E: java.lang.NullPointerException: Attempt to invoke interface method 'void com.github.adamantcheese.chan.core.image.ImageListener.onErrorResponse(com.android.volley.VolleyError)' on a null object reference
        at com.github.adamantcheese.chan.core.image.ImageLoaderV2.lambda$null$0(ImageLoaderV2.java:117)
        at com.github.adamantcheese.chan.core.image.-$$Lambda$ImageLoaderV2$9FvfVKiM5vb6ewdwdUs4HVbO27g.run(Unknown Source:4)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7343)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:933)
2019-07-30 12:52:23.855 E: .
```

Haha. Apparently I forgot to put ImageListener into ImageContainer. Thank god this shit is not merged yet. This bug breaks the app to the point that you have to clear all the data.

To reproduce: 
- Download a thread.
- Delete couple of images from the /images/ directory.
- Open local thread.

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/image/ImageLoaderV2.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                        postImage.spoilerThumbnailUrl.toString());
+
+                formattedName = ThreadSaveManager.formatSpoilerImageName(extension);
+            } else {
+                if (isThumbnail) {
+                    String extension = StringUtils.extractFileExtensionFromImageUrl(
+                            postImage.thumbnailUrl.toString());
+
+                    if (extension == null) {
+                        Logger.e(TAG, "Could not extract extension from a thumbnailUrl = "
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/image/ImageLoaderV2.java]: Probably should throw an exception here or just return an empty image container.

**Comment by Adamantcheese** [comment]
Tell me when this is good to merge then.

**Comment by K1rakishou** [comment]
Sure.

**Comment by K1rakishou** [comment]
@Adamantcheese 
Should be okay to merge this.

**Comment by Adamantcheese** [comment]
Alright, will do today or tomorrow.

**Pull request merged**